### PR TITLE
[Doc] Fix friend trees example of RDF

### PR DESCRIPTION
**Pull request by stwunsch**
@bluehood Do I understand this correctly?

### CODE REVIEW

**Comment by etejedor** [comment]
Yes, the alias should not be necessary and you should be able to use the `myFriend.MyCol` syntax in the `Filter`, thanks to [this](https://github.com/root-project/root/blob/master/tree/dataframe/src/RDFInterfaceUtils.cxx#L143).

**Comment by stwunsch** [comment]
@bluehood @etejedor I've fixed the docs. Does the current section hold true?

File: tree/dataframe/src/RDataFrame.cxx [hunk 1]
```diff
@@ -501,19 +501,15 @@ You see how we created one `double` variable for each thread in the pool, and la
 Friend trees are supported by RDataFrame.
 In order to deal with friend trees with RDataFrame, the user is required to build
 the tree and its friends and instantiate a RDataFrame with it.
-Two caveats are presents when using jitted `Define`s and `Filter`s:
-1) the only columns which can be used in the strings passed to the aforementioned transformations are the top level branches of the friend trees.
-2) the "friend columns" cannot be written with the notation involving a dot. For example, if a tree is created like this:
+One caveat is present when using jitted `Define`s and `Filter`s:
+The only columns which can be used in the strings passed to the aforementioned transformations are the top level branches of the friend trees.
```
Comment (etejedor) [inline tree/dataframe/src/RDataFrame.cxx]: This is not true anymore, you can pass the same depth of branch names with friend trees as with the main tree. So, you can do "myFriend.myBranch.mySubBranch.myLeaf"

Comment (stwunsch) [reply tree/dataframe/src/RDataFrame.cxx]: ok, i'll adapt it.

**Comment by stwunsch** [comment]
@etejedor @bluehood feel free to hit merge if you are satisfied.

**Comment by bluehood** [comment]
@stwunsch sorry for not being too responsive.
the PR looks good to me and it should solve a recent jira ticket, [ROOT-9546](https://sft.its.cern.ch/jira/browse/ROOT-9546).

**Pull request merged**